plugins/pluginmanagers/lazy.nix

```diff
@@ -203,13 +213,18 @@ in
       in
       mkIf (cfg.plugins != [ ]) ''
         require('lazy').setup(
-          {
-            dev = {
-              path = "${lazyPath}",
-              patterns = {"."},
-              fallback = false
-            },
-            spec = ${helpers.toLuaObject packedPlugins}
+          ${
+            helpers.toLuaObject (
+              {
+                dev = {
+                  path = "${lazyPath}";
+                  patterns = [ "." ];
+                  fallback = false;
+                };
+                spec = packedPlugins;
+              }
+              // cfg.setup
+            )
```

Comment:
Instead of updating the final value here, we would usually assign these values to `plugins.lazy.settings` using the module system. That way we can take advantage of things like `mkDefault`, `mkIf`, etc.

Reply:
`settingsOptions` should only contain sub-options that correspond exactly to what the upstream plugin expects in its `setup({})` table.

If we need to define additional options (either internal or user-facing), they can be added to `extraOptions` instead (and will sit _alongside_ `settings`).

Reply:
Essentially, the settigns option iteslf should usually serve the role being served here by `setupOptions`.

Reply:
> Or because dev itself is a sub table it should still be in setupOptions?

We shouldn't have a `setupOptions` at all.

> dev is actually in the setup({}) table

Gotcha, wasn't sure based on your earlier example :slightly_smiling_face: